spec/models/manageiq/providers/ibm_power_hmc/infra_manager/refresher_spec.rb

```diff
@@ -44,11 +44,46 @@ def assert_specific_vm
         :build_number => "7300-00-00-0000"
       )
     end
+  end
 
-    def full_refresh(ems)
-      VCR.use_cassette(described_class.name.underscore) do
-        EmsRefresh.refresh(ems)
+  context "#target_refresh" do
+    let(:ems) do
+      ext = FactoryBot.create(:ems_ibm_power_hmc_infra_with_authentication)
+      full_refresh(ext)
+      ext
+    end
+
+    def target_refresh(target, example)
+      expect(target).not_to be_nil
+      VCR.use_cassette("#{described_class.name.underscore}_#{example.description}") do
+        EmsRefresh.refresh(target)
       end
     end
+
+    it "host" do |example|
+      target_refresh(ems.hosts.first, example)
```

Comment:
Is that OK if the idea is to trigger targeted refresh on any host just to ensure coverage for the targeted collection path and avoiding assumptions on the actual available hosts ? I may as well have used `.sample` instead of `.first`.

Reply:
Well the issue is if you're using a VCR and you pass in a different host the VCR will fail because it won't have the correct API requests to replay

Reply:
Ok, makes sense